### PR TITLE
fix: update cli yaml files to reflect auth LW12 features

### PR DESCRIPTION
**Pull request by J0**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Update the CLI YAML config for Auth with the configuration for recent MFA, Third Party Auth, and Anonymous Sign In changes so that the documentation displays on our site and users are aware of the features

Referencing [this example file](https://github.com/supabase/cli/blob/15833d1896185018c343352cc1d1f38d4de83ecb/pkg/config/testdata/config.toml#L4) from CLI

We will add session control and hooks in a FLUP

### CODE REVIEW

File: apps/docs/spec/cli_v1_config.yaml [hunk 1]
```diff
@@ -1112,6 +1180,17 @@ parameters:
       - name: 'Auth Server configuration'
         link: 'https://supabase.com/docs/reference/auth'
 
+  - id: 'auth.third_party.aws_cognito'
```
Comment (J0) [inline apps/docs/spec/cli_v1_config.yaml:1183]: can consider `<provider>` syntax, but let's wait for verdict of [this internal discussion](https://supabase.slack.com/archives/C023E4L60R3/p1733205517257259)

File: apps/docs/spec/cli_v1_config.yaml [hunk 2]
```diff
@@ -1090,6 +1114,50 @@ parameters:
...
+    tags: ['auth', 'local']
+    required: false
+    default: 'false'
+    description: |
+      Enable Phone enrollment for multi-factor authentication.
+    links:
+      - name: 'Auth Server configuration'
+        link: 'https://supabase.com/docs/reference/auth'
+
+  - id: 'auth.mfa.phone.verify_enabled'
```
Comment (J0) [inline apps/docs/spec/cli_v1_config.yaml:1150]: ditto with below, can consider <provider> syntax

File: apps/docs/spec/cli_v1_config.yaml [hunk 3]
```diff
@@ -636,9 +636,31 @@ parameters:
       - name: 'Auth Server configuration'
         link: 'https://supabase.com/docs/reference/auth'
 
+  - id: 'auth.enable_anonymous_sign_ins'
+    title: 'auth.enable_anonymous_sign_ins'
+    tags: ['auth', 'local']
+    required: false
+    default: 'true'
```
Comment (J0) [inline apps/docs/spec/cli_v1_config.yaml:643]:
```suggestion
    default: 'false'
```

File: apps/docs/spec/cli_v1_config.yaml [hunk 4]
```diff
@@ -636,11 +636,33 @@ parameters:
...
+    links:
+      - name: 'Auth Server configuration'
+        link: 'https://supabase.com/docs/reference/auth'
+
   - id: 'auth.enable_refresh_token_rotation'
     title: 'auth.enable_refresh_token_rotation'
     tags: ['auth']
     required: false
-    default: 'true'
+    default: 'false'
```
Comment (J0) [inline apps/docs/spec/cli_v1_config.yaml:665]:
```suggestion
    default: 'true'
```

File: apps/docs/spec/cli_v1_config.yaml (hunk 2 above)
Comment (J0) [reply apps/docs/spec/cli_v1_config.yaml:1194]: Probably not on further thought, realized they have distinct fields

**Review by charislam — APPROVED** [review]

**Pull request merged**